### PR TITLE
Improve Podman provider

### PR DESCRIPTION
**Pull request by abbra**
I needed these changes to get FreeIPA topologies working with mrack podman provider.

You can see a sample configuration in https://github.com/abbra/freeipa-local-tests/, where I am able to set up two parallel IPA deployments in the same topology metadata and then establish trust between those IPA environments (using work-in-progress COPR).

### CODE REVIEW

File: src/mrack/data/provisioning-config.yaml [hunk 1]
```diff
@@ -21,17 +24,7 @@ podman:
...
-        # Mount a temporary filesystems (tmpfs) into a container
-        "--tmpfs":
-            - "/tmp"
-            - "/run"
-            - "/run/lock"
-        # Use /sys/fs/cgroup in container as read only volume
-        "-v":
-            - "/sys/fs/cgroup:/sys/fs/cgroup:ro"
-        # Adding ipv6 support to network
-        "--network": "enable_ipv6=true"
```
Comment (dav-pascual) [inline src/mrack/data/provisioning-config.yaml:34]: This file is just a means for documentation and serves as an example. It doesn't affect execution. I would probably only remove the last line

File: src/mrack/providers/podman.py [hunk 2]
```diff
@@ -173,7 +179,8 @@ async def wait_till_provisioned(self, resource):
 
         while datetime.now() < timeout_time:
             try:
-                server = await self.podman.inspect(cont_id)[0]
+                servers = await self.podman.inspect(cont_id)
+                server = servers[0]
```
[the remark this answers is not in the document]
Comment (f-trivino) [reply src/mrack/providers/podman.py:183]: This change fixes:

```
  File "/home/ftrivino/.local/lib/python3.12/site-packages/mrack/providers/podman.py", line 176, in wait_till_provisioned
    server = await self.podman.inspect(cont_id)[0]
                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^
TypeError: 'coroutine' object is not subscriptable
```

[the remark this answers is not in the document]
Comment (dav-pascual) [reply src/mrack/providers/podman.py:183]: gotcha

File: src/mrack/transformers/podman.py [hunk 3]
```diff
@@ -29,6 +29,7 @@ class PodmanTransformer(Transformer):
         "images",
         "pubkey",
         "default_network",
+        "network_options",
```
[the remark this answers is not in the document]
Comment (f-trivino) [reply src/mrack/transformers/podman.py:32]: I guess that's ok then, as long as podman provisioner works when defining network options.

**Comment by dav-pascual** [comment]
Amending changes in https://github.com/neoave/mrack/pull/300

**Comment by dav-pascual** [comment]
#300 merged

**Pull request closed without merging**